crypto/src/hash/rescue_prime/mod.rs

```diff
@@ -0,0 +1,816 @@
+use crate::alloc::vec::Vec;
+use core::iter;
+use lambdaworks_math::field::element::FieldElement;
+use lambdaworks_math::field::fields::u64_goldilocks_field::Goldilocks64Field;
+
+mod parameters;
+mod utils;
+
+use parameters::*;
+use utils::*;
+
+type Fp = FieldElement<Goldilocks64Field>;
+
+#[derive(Clone)]
+#[allow(dead_code)]
+enum MdsMethod {
+    MatrixMultiplication,
+    Ntt,
+    Karatsuba,
+}
+#[warn(dead_code)]
+pub struct RescuePrimeOptimized<const SECURITY_LEVEL: usize, const NUM_FULL_ROUNDS: usize> {
```

Comment:
Is it strictly necessary for the `SECURITY_LEVEL` to be a const generic? Can't we have a `security_level: SecurityLevel` field in `RescuePrimeOptimized`? This would simplify some logic and help remove the `assert` in `RescuePrimeOptimized::new`. If we cannot, other ways exist to simplify said logic.

From what I could see from this code, this is not enough justified (but I could be missing some context).